### PR TITLE
Add type stubs for org.optaplanner and java packages

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**
Currently no type infomation is displayed when interacting with optapy objects (ConstraintFactory, SolverConfig, etc.) in an IDE, which is a terrible experience. Using stubgenj allows us to add this type information without needing to copy it from java, massively improving the IDE experience.

### CODE REVIEW

File: optapy-core/setup.py [hunk 1]
```diff
@@ -52,8 +75,27 @@ def run(self):
         build_py.run(self)
 
 
+def find_stub_files(stub_root: str):
+    """
+    This function is taken from the awesome sqlalchey-stubs:
+    https://github.com/dropbox/sqlalchemy-stubs/blob/master/setup.py#L32
+    It's licensed under Apache 2.0:
+    https://github.com/dropbox/sqlalchemy-stubs/blob/master/LICENSE
+    """
```
Comment (triceo) [inline optapy-core/setup.py:84]: As much as I appreciate the spirit of fair play, this code is not rocket science. It is a fairly generic piece of code anyone could have written, and in fact probably has. I wouldn't bother with credit in this particular case.

File: optapy-core/pom.xml [hunk 2]
```diff
@@ -71,6 +71,23 @@
...
+      <activation>
+        <property>
+          <name>stubs</name>
+        </property>
+      </activation>
+      <dependencies>
+        <dependency>
+          <groupId>org.kie.kogito</groupId>
+          <artifactId>kogito-legacy-api</artifactId>
+        </dependency>
```
Comment (triceo) [inline optapy-core/pom.xml:86]: Do we actually reference KIE API in our APIs anywhere?
(My guess is it leaked into `config`. That said, exposing foreign APIs is a bad practice. We should avoid that.)

Comment (Christopher-Chianelli) [reply optapy-core/pom.xml:86]: It in `ScoreDirectorConfig`; without it, an infinite loop occurs in stubgenj (I think the bug was recently patched) and the ScoreDirectorConfig stub is not generated.

File: optapy-core/setup.py (hunk 1 above)
Comment (Christopher-Chianelli) [reply optapy-core/setup.py:84]: TBH, I copied it from another project, which copied it from that project (with the above comment). Referencing the source can help with context when re-reading the code in the future. Although the code is fairly simple.

File: optapy-core/pom.xml (hunk 2 above)
Comment (Christopher-Chianelli) [reply optapy-core/pom.xml:86]: Looking at ScoreDirectorConfig, it does not reference KieRuntimeBuilder, yet this error occurs:
```
Skipping <java class 'org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig'> due to java.lang.NoClassDefFoundError: Lorg/kie/kogito/legacy/rules/KieRuntimeBuilder;
reference to missing class ScoreDirectorFactoryConfig - generating empty stub
```

it does not go into an infinite loop, however; the only class not generated is the ScoreDirectorFactoryConfig if we remove `kogito-legacy-api` from the stubs dependencies.

**Pull request merged**